res/css/views/rooms/_RoomHeader.pcss

```diff
@@ -87,3 +87,9 @@ limitations under the License.
         background: var(--cpd-color-bg-subtle-primary);
     }
 }
+.mx_RoomHeader_ChatIcon {
+    * {
+        fill: $icon-button-color;
+    }
```

Comment:
it seems like it would be better to update the SVG to have `fill="currentColor"` than to forcibly override it here. Generally, I don't think it should be necessary to have a dedicated CSS class. (There isn't one for the "threads" button, for example.)

Reply:
@manancodes have you had a chance to look at this?

Reply:
I looked into it again. The other icons being used are imported from the [vector-im/compound-design-tokens](https://github.com/vector-im/compound-design-tokens) library. Which is why it does not require any additional css.
<img width="501" alt="Screenshot 2023-11-07 at 8 00 34 PM" src="https://github.com/matrix-org/matrix-react-sdk/assets/54790263/75fd8369-aada-4f70-ad17-e7442b876076">

And the chat icon present there is an outline icon, which does not go well with the other icons. 
https://github.com/vector-im/compound-design-tokens/blob/main/icons/chat.svg

In the legacy header, the SVG is used as a mask and background color property is added to the icon.
https://github.com/matrix-org/matrix-react-sdk/blob/6ae7c033d572fcb37a191a7cf61634c85b8dbaf9/res/css/views/rooms/_LegacyRoomHeader.pcss#L187

So I used that Icon and applied css to have dynamic colors according to the selected theme. What is the right way to tackle this? Should the filled chat icon be added in the Library? What is the procedure for that?

Reply:
Good question. Yes, it definitely looks like we should be using a Compound design icon here, since all the other buttons in the header use one. And you're right that there only seems to be an outline icon there at the moment.

@kerryarchibald as our Compound Champion, can you advise on the best approach here?